modules/frontend/combiner/metrics_query_range.go

```diff
@@ -88,6 +96,67 @@ func sortResponse(res *tempopb.QueryRangeResponse) {
 	}
 }
 
+// diffResponse takes two QueryRangeResponses and returns a new QueryRangeResponse that contains only the differences between the two.
+// it creates a completely new response, so the input responses are not modified. an in place diff would be nice for memory savings, but
+// the diffResponse is returned and marshalled into proto. if we modify an object while it's being marshalled this can cause a panic
+func diffResponse(prev, curr *tempopb.QueryRangeResponse) *tempopb.QueryRangeResponse {
+	if prev == nil {
+		return curr
+	}
+
+	seriesIdx := 0
+	diff := &tempopb.QueryRangeResponse{
+		Series: make([]*tempopb.TimeSeries, 0, len(curr.Series)/2), // prealloc half of series. untuned
+	}
+
+	// series are sorted, so we can do this in a single pass
+	for _, s := range curr.Series {
+		// is this a series that's new in curr that wasn't in prev? this check assumes that
+		// a series can not be removed from the output as the input series are combined
+		if seriesIdx >= len(prev.Series) || s.PromLabels != prev.Series[seriesIdx].PromLabels {
+			diff.Series = append(diff.Series, s)
+			continue
+		}
+
+		// promlabels are the same, have to check individual samples
+		// just copy forward labels and exemplars
+		diffSeries := &tempopb.TimeSeries{
+			Labels:     s.Labels,
+			PromLabels: s.PromLabels,
+			Exemplars:  s.Exemplars,                                  // taking all current exemplars. improve?
```

Comment:
It probably makes sense to diff the exemplars.  Is there a reason not to?  They should be stable, but even if they aren't, there would still be some savings.

Reply:
i wasn't sure about exemplars so i did the most straightforward thing. will diff them as well 👍